### DISCUSSION TITLE
Is there a process for getting a new linear solver integrated?

### DISCUSSION THREAD

**Discussion by mahonbt1** [coin-or/Ipopt#630] [Q&A] [answered]
I'd be curious to see how Dr Tim Davis's [SuiteSparse](https://github.com/DrTimothyAldenDavis/SuiteSparse) library holds up as the backbone for Ipopt.  I think he builds his stuff without fortran for the most part, which could simplify the build dependencies a bit.  If there is a process to do this work documented somewhere I might look into it.

**Comment by svigerske** [comment, marked as answer]
The easiest is probably to start with a separate project for the Ipopt/SuiteSparse interface and to link against an Ipopt library. As a library user, you can provide a linear solver when initializing Ipopt: https://github.com/coin-or/Ipopt/discussions/618

Alternatively, one would edit Ipopt itself by copying the logic for one of the existing linear solvers (maybe Spral or Wsmp). Next to providing the SuiteSparse interface class, this would also require to update the buildsystem and AlgorithmBuilder class.

**Reply by damienhocking** [reply]
I did the MUMPS interface many years ago, that one is pretty simple too.
